File: Finished Projects (JAVA)/TMairella_numberguessinggame_v1.java
import java.util.*; 

public class TMairella_numberguessinggame_v1{ 
    public static void main(String[] args) {
        
        int randomNum = (int)(Math.random() * 100) + 1;
        int userguess = 0;
        int attempts = 0;
        Scanner input = new Scanner(System.in);
      
        System.out.println("Welcome to the Number Guessing Game!");
        System.out.println("I have selected a random number between 1 and 100.");
        System.out.println("Try to guess the number, and I will tell you if it's higher or lower.");

        while (userguess != randomNum) {
            System.out.print("Enter your guess: ");
            userguess = input.nextInt();
            attempts++;
            if (userguess > randomNum) {
                System.out.println("Lower (Attempts so far: " + attempts + ")");
            } else if (userguess < randomNum) {
                System.out.println("Higher (Attempts so far: " + attempts + ")");
            } else {
                System.out.println("Congrats, You got the number right! The number was " + randomNum);
                System.out.println("You got the correct number in " + attempts + " attempts!");
            }
        }
        input.close();
     }
} 

File: Finished Projects (PYTHON)/TMairella_numberguessinggame_v1.py
import random

play_again = "yes"
print("Welcome to the Number Guessing Game!")

while play_again:
    randomNum = random.randint(1, 100)
    attempts = 0

    print("I have selected a number between 1 and 100. Can you guess it?")

    while True:
        guess = int(input("Enter your guess: "))
        attempts += 1

        if guess < randomNum:
            print("Too low! Try again.")
        elif guess > randomNum:
            print("Too high! Try again.")
        else:
            print("Congratulations! You've guessed the number!")
            print(f"You got the correct number in {attempts} attempts!")
            break

    play_again = input("Would you like to play again? (yes/no): ").lower()

    if play_again != "yes":
        print("Thanks for playing! Goodbye!")
        break
